Fix type field

bilalebi opened the pull request:
This API call: https://resolver.ensembl.org/id/ENSAROG00010015245
Causes the resolver to throw:
```
│ pydantic_core._pydantic_core.ValidationError: 1 validation error for ResolvedPayload                                                                                                                        
│ type                                                                                                                                                                                                        
│  Input should be a valid string [type=string_type, input_value={'kind': 'strain', 'value': 'reference'}, input_type=dict]
```

Because it's expecting `type` field to be a `string` but it's getting a dictionary `{'kind': 'strain', 'value': 'reference'}`

To fix it, I changed
```
type: Optional[str] = None
# to
type: Optional[Dict[str, str]] = None
```

azangru commented:
If the resolver finds multiple matches, it renders an html page where it shows a list of them, with links.

What can now be seen on this html page is that:
- The type information gets rendered as a stringified json
- It also is placed in the `Resolved URL` column rather than in the `Type` column, where it belongs

<img width="1073" height="166" alt="image" src="https://github.com/user-attachments/assets/d29d0117-2480-4fad-a7c1-cbea618fa749" />

This calls for an update in this html file:

https://github.com/Ensembl/ensembl-web-resolver/blob/main/app/api/resources/templates/search_results.html#L24

bilalebi commented:
Fixed now and added `is_reference` logic as well
Here is the example that was broken: https://resolver.ensembl.org/id/ENSAROG00010015245
<img width="1968" height="286" alt="image" src="https://github.com/user-attachments/assets/bea43930-4c62-4ca3-80e6-e9601fbd5967" />

Another example that was working but had None in Type: https://resolver.ensembl.org/id/ENSG00000127720
<img width="1954" height="336" alt="image" src="https://github.com/user-attachments/assets/cbefe7f9-2655-479c-a8a4-d4c3c276a68d" />
